test/jdk/java/nio/channels/FileChannel/BlockDeviceSize.java

```diff
@@ -24,41 +24,47 @@
 /* @test
  * @bug 8054029 8313368
  * @requires (os.family == "linux")
- * @summary Block devices should not report size=0 on Linux
- * @run main/manual BlockDeviceSize
+ * @summary FileChannel.size() should be equal to RandomAccessFile.size() and > 0 for block devs on Linux
  */
 
 import java.io.RandomAccessFile;
 import java.nio.file.Path;
-import java.nio.file.Paths;
 import java.nio.channels.FileChannel;
 import java.nio.file.AccessDeniedException;
 import java.nio.file.NoSuchFileException;
+import java.util.List;
+
 import static java.nio.file.StandardOpenOption.*;
 
 
 public class BlockDeviceSize {
-    private static final String BLK_FNAME = "/dev/sda1";
-    private static final Path BLK_PATH = Paths.get(BLK_FNAME);
+    private static final List<String> BLK_FNAMES = List.of("/dev/sda1", "/dev/nvme0n1", "/dev/xvda1") ;
 
     public static void main(String[] args) throws Throwable {
-        try (FileChannel ch = FileChannel.open(BLK_PATH, READ);
-             RandomAccessFile file = new RandomAccessFile(BLK_FNAME, "r")) {
+        for (String blkFname: BLK_FNAMES) {
+            Path blkPath = Path.of(blkFname);
+            try (FileChannel ch = FileChannel.open(blkPath, READ);
+                 RandomAccessFile file = new RandomAccessFile(blkFname, "r")) {
+
+                long size1 = ch.size();
+                long size2 = file.length();
+                if (size1 != size2) {
+                    throw new RuntimeException("size differs when retrieved" +
+                            " in different ways: " + size1 + " != " + size2);
+                }
+                if (size1 <= 0) {
+                    throw new RuntimeException("size() for a block device size returns zero or a negative value");
+                }
+                System.out.println("OK");
 
-            long size1 = ch.size();
-            long size2 = file.length();
-            if (size1 != size2) {
-                throw new RuntimeException("size differs when retrieved" +
-                        " in different ways: " + size1 + " != " + size2);
+            } catch (NoSuchFileException nsfe) {
+                System.err.println("File " + blkFname + " not found." +
+                        " Skipping test");
+            } catch (AccessDeniedException ade) {
+                throw new RuntimeException("Access to " + blkFname + " is denied."
```

Comment:
Looking at it again, this should be SkipException rather RuntimeException, otherwise the test will fail when denied.